src/accelerate/utils/modeling.py

```diff
@@ -143,7 +143,12 @@ def set_module_tensor_to_device(
         elif value is not None or torch.device(device) != module._parameters[tensor_name].device:
             param_cls = type(module._parameters[tensor_name])
             kwargs = module._parameters[tensor_name].__dict__
-            new_value = param_cls(new_value, requires_grad=old_value.requires_grad, **kwargs).to(device)
+            if (kwargs.get("ds_tensor") is not None):
```

Comment:
So let's make this check a bit stronger, like the `param_cls.__name__` is bnb8 parameter class or not, since we only want to pass the kwargs for them.

Reply:
Yea, no problem.

Reply:
FYI the bnb parameter name is `Int8Params` ;) !